test/integration_test.py

```diff
@@ -745,18 +750,24 @@ def _test_repository_files(self, catalog: str):
                     response = self._get_url_json(response['Location'])
 
                 response = self._get_url(response['Location'], stream=True)
-                self._validate_fastq_response(response)
+                self._validate_file_response(response, file['format'])
 
-    def _validate_fastq_response(self, response: urllib3.HTTPResponse):
+    def _validate_file_content(self, content: ReadableFileObject, format: str):
+        if format == 'fastq.gz':
```

Comment:
I don't think we can trust the format and should instead base this on the extension in the file name, after converting it to lower-case.